README.md
# JobControl is no longer maintained! If you'd like to take over please [contact me](https://github.com/kevinrenskers).

## Overview
JobControl (dmmjobcontrol) is a TYPO3 extension for showing jobs (“vacancies”) on your website. It provides a list- and detail view and the ability to search and apply for jobs. It can even make RSS feeds of your jobs.

It works with html templates so it's easy to configure how the extension will look for your site. The list can be shown as a “paginated list”, including a page-browser. The extension itself is multi-lingual, at this moment English, Danish, Polish, German, Russian and Dutch are included. The best feature however is that multi-lingual jobs are fully supported too, so you can provide a translation for a job if you have a multi-lingual site.

JobControl uses MM-relation tables for regions, branches, sectors etc. This means that for every new site, you can make a new list of branches to use. They are not hardcoded and don't require any TypoScript to set up.

JobControl is very easy to set up, with good default templates that can be styled to your needs using css stylesheets. It's very powerful and flexible too with lots of configuration options for advanced users.

JobControl was [originally created in 2007](http://mixedcase.nl/articles/2013/04/10/dmmjobcontrol-six-years-old/) at DMM Websolutions in Groningen, The Netherlands.

### Features
* Paginated lists
* List recent jobs
* Multi-lingual job details with apply form
* Search view
* RSS feeds
* Powerful, flexible and easily extendible
* Tested with TYPO3 4.5, 4.6, 4.7 and 6.0

## Documentation
You can read the documentation online at http://mixedcase.nl/projects/dmmjobcontrol/.

## License
JobControl is free software under the terms of the
GNU General Public License version 2 or any later version.
The GNU/GPL license (version 2) is found in the file LICENSE.

# Contributors wanted!
As I don't use PHP or TYPO3 myself anymore, it's becoming a bit of a burden to support JobControl and release bugfixes, support new versions of TYPO3, etc. If you'd like to support this open source project please feel free to fork the code and tackle any open issues, fix bugs, support the new TYPO3 features, add translations, or anything else you can think of. Push access will be given to regular contributors.

If you'd like to adopt JobControl please [contact me](https://github.com/kevinrenskers).


CHANGELOG
2015-01-11 Kevin Renskers [mixedCase]
* Version 2.16.0
* Marked the extension as obsolete. It's been a long wild ride, but I can't support 
  this extension anymore. If you'd like to take over please contact me at info@mixedcase.nl.

2014-10-17 Kevin Renskers [mixedCase]
* Version 2.15.0
* Security bugfix, thank you weber-ebusiness for the pull request!
* 6.2 LTS compatibilty
* Enable backend search

2013-08-25 Kevin Renskers [mixedCase]
* Version 2.14.0
* Disabled all caching, essentially going back to the behaviour of 2.11.0 and before.
* Contributions to properly deal with caching are greatly appreciated!

2013-07-30 Kevin Renskers [mixedCase]
* Version 2.13.0
* Bugfix: prevent double apply form submits when including both the detail plugin and the apply plugin on the same page
* Bugfix: prevent caching of the detail page when the apply form was submitted

2013-07-11 Kevin Renskers [mixedCase]
* Version 2.12.1
* Bugfix: links to detail pages were broken when "road 1" was used

2013-07-09 Kevin Renskers [mixedCase]
* Version 2.12.0
* Bugfix: links to detail pages now work in TYPO3 6.1

2013-05-09 Kevin Renskers [mixedCase]
* Version 2.11.0
* Feature: it's now possible to extend the apply form
  (thanks Heller und Kuschel GbR for sponsoring the development)

2013-04-11 Kevin Renskers [mixedCase]
* Version 2.10.0
* Feature: added ###TSTAMP### marker and ###TSTAMP_LABEL### label
* Change: documentation moved to http://mixedcase.nl/projects/dmmjobcontrol/
* Change: removed the XLIFF language files until I can figure out how to support
  this format without duplicating all my work.

2013-04-09 Kevin Renskers [mixedCase]
* Version 2.9.0
* Feature: it's now possible to extend the available (multi-lingual) labels
  via the labelArrayFunction. See documentation and example labelArrayFunction.php.
* Note, this deprecates the non-documented labelMarkersFunction feature.

2013-04-06 Kevin Renskers [mixedCase]
* Version 2.8.1
* Bugfix: the changes in 2.8.0 caused file upload fields to no longer be required.
  Please note that the fix caused a backwards incompatible change in the default apply form.
  Check the supplied default apply.tmpl for the new usage of file upload fields if you're
  using a custom apply form.

2013-04-06 Kevin Renskers [mixedCase]
* Version 2.8.0
* Feature: it's now possible to have any number of file upload fields in the apply form
  (thanks Klaus Prochart for sponsoring the development of this feature)
* Feature: the list of allowed file extensions can now be configured in typoscript
* Bugfix: fixed link to attachment in plain text email

2013-04-03 Kevin Renskers [mixedCase]
* Version 2.7.4
* Bugfix: removed error_reporting(E_STRICT) accidentally left in (thanks Jan Schreier)

2013-04-03 Kevin Renskers [mixedCase]
* Version 2.7.3
* Bugfix: fixed the included typoscript setup (thanks Klaus Prochart)

2013-04-02 Kevin Renskers [mixedCase]
* Version 2.7.2
* Change: better explanation of the "multiple" options in the constant editor
* Change: 'property_sort' is replaced by the 'sort.' array, making it possible to set
  a different sort config per property.

2013-03-31 Kevin Renskers [mixedCase]
* Version 2.7.1
* Bugfix: changed default search form to have the reset button as the second button,
  solving the problem that submitting with the enter button resets the search query
* Bugfix: solved the "having no baseUrl breaks redirects" problem
* Thanks Philipp Kitzberger for reporting these problems

2013-03-31 Kevin Renskers [mixedCase]
* Version 2.7.0
* Feature: it's now possible to set list sorting in the url like this:
  ?tx_dmmjobcontrol_pi1[sort]=region&tx_dmmjobcontrol_pi1[sort_order]=asc
  (thanks Heiko Haensge)

2013-03-31 Kevin Renskers [mixedCase]
* Version 2.6.0
* Feature: added possibility to sort properties like region, category and sector in the backend
  Usage: plugin.tx_dmmjobcontrol_pi1.property_sort = sorting ASC
* Feature: the list template can now have TEMPLATE_NO_JOBS and TEMPLATE_JOBS subparts
  (see default list.tmpl for example)
* Change: fixed some minor code style related things
* Change: in the backend the jobs table is now shown before all the others
* Thanks Klaus Prochart for sponsoring the development of these features

2013-03-11 Kevin Renskers [mixedCase]
* Version 2.5.3
* Extension was tested in TYPO3 6.0, no problems were found
* Bugfix: fixed several PHP warnings (thanks Heiko Haensge)
* Change: added docx and odt to allowed file extensions (thanks Jörgen van der Kroon)

2012-08-21 Kevin Renskers [mixedCase]
* Version 2.5.2
* Bugfix: fixed sending mail (apply form)

2012-08-19 Kevin Renskers [mixedCase]
* Version 2.5.1
* Change: updated README and documentation

2012-08-15 Kevin Renskers [mixedCase]
* Version 2.5.0
* JobControl is back and on GitHub (https://github.com/kevinrenskers/dmmjobcontrol)!
* Bugfix: now TYPO3 4.5, 4.6 and 4.7 compatible
* Bugfix: fixed selectboxes in search
* Bugfix: fixed localization of database values for TYPO3 4.6 and 4.7
* Change: moved localization files to XLIFF format

2010-07-16 Jan Wolters [DMM Websolutions]
* Version 2.4.1
* Change: author information updated

2010-07-16 Jan Wolters [DMM Websolutions]
* Version 2.4.0
* Feature: in the TypoScript code you can now use the new config option "ignorehidden"
  Using this feature you can show hidden vacancies for testing
* Feature: in the Typoscript code you can now use the new config option "apply.redirect"
  Using this feature you can redirect a user for apply-confirmation to a TYPO3 page
* Bugfix: XML-feed will use encoding defined for page (page.config.renderCharset)

2010-05-21 Jan Wolters [DMM Websolutions]
* Version 2.3.8
* Feature: in the TypoScript code you can now use the new config option "multipleselect"
  Using this feature you can use a multiple select in the search for the fields region,

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "dmmjobcontrol".
 *
 * Auto generated 25-08-2013 11:50
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array(
	'title' => 'JobControl',
	'description' => 'Job database to show job offers on your site. Paginated lists, detailview, search function, apply form and RSS. It is possible to show the most recent jobs and/or the searchform on all pages. Easy to setup yet powerful and flexible. Please ask for free support on the mailing lists, or contact the owner for paid support.',
	'category' => 'plugin',
	'shy' => 0,
	'version' => '2.16.0',
	'dependencies' => 'cms',
	'conflicts' => '',
	'priority' => '',
	'loadOrder' => '',
	'module' => '',
	'state' => 'obsolete',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearcacheonload' => 0,
	'lockType' => '',
	'author' => 'Kevin Renskers',
	'author_email' => '',
	'author_company' => '',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
			'typo3' => '4.5.0-6.2.0',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:52:{s:7:"AUTHORS";s:4:"ee21";s:9:"CHANGELOG";s:4:"dd71";s:19:"de.locallang_db.xlf";s:4:"69f0";s:20:"de.locallang_tca.xlf";s:4:"6855";s:19:"dk.locallang_db.xlf";s:4:"8f51";s:20:"dk.locallang_tca.xlf";s:4:"747d";s:12:"ext_icon.gif";s:4:"b6b0";s:17:"ext_localconf.php";s:4:"440c";s:14:"ext_tables.php";s:4:"7a2a";s:14:"ext_tables.sql";s:4:"bc7d";s:28:"ext_typoscript_constants.txt";s:4:"d28e";s:24:"ext_typoscript_setup.txt";s:4:"e4b8";s:15:"flexform_ds.xml";s:4:"7c17";s:31:"icon_tx_dmmjobcontrol_arrow.gif";s:4:"3d4b";s:34:"icon_tx_dmmjobcontrol_category.gif";s:4:"3073";s:33:"icon_tx_dmmjobcontrol_contact.gif";s:4:"3073";s:36:"icon_tx_dmmjobcontrol_discipline.gif";s:4:"3073";s:35:"icon_tx_dmmjobcontrol_education.gif";s:4:"3073";s:29:"icon_tx_dmmjobcontrol_job.gif";s:4:"5e8d";s:32:"icon_tx_dmmjobcontrol_job__h.gif";s:4:"9f08";s:33:"icon_tx_dmmjobcontrol_job__ht.gif";s:4:"4a78";s:32:"icon_tx_dmmjobcontrol_job__t.gif";s:4:"ea8f";s:32:"icon_tx_dmmjobcontrol_job__x.gif";s:4:"5e8d";s:32:"icon_tx_dmmjobcontrol_region.gif";s:4:"3073";s:32:"icon_tx_dmmjobcontrol_sector.gif";s:4:"3073";s:7:"LICENSE";s:4:"c78c";s:16:"locallang_db.xlf";s:4:"378d";s:16:"locallang_db.xml";s:4:"1f52";s:17:"locallang_tca.xlf";s:4:"f239";s:17:"locallang_tca.xml";s:4:"faff";s:19:"nl.locallang_db.xlf";s:4:"9eb8";s:20:"nl.locallang_tca.xlf";s:4:"17f7";s:19:"pl.locallang_db.xlf";s:4:"3d7f";s:20:"pl.locallang_tca.xlf";s:4:"da26";s:9:"README.md";s:4:"9aa5";s:19:"ru.locallang_db.xlf";s:4:"bf14";s:7:"tca.php";s:4:"22df";s:14:"doc/manual.sxw";s:4:"4ab2";s:19:"doc/wizard_form.dat";s:4:"5794";s:20:"doc/wizard_form.html";s:4:"def0";s:34:"pi1/class.tx_dmmjobcontrol_pi1.php";s:4:"d076";s:17:"pi1/locallang.xml";s:4:"a167";s:24:"pi1/static/editorcfg.txt";s:4:"08cd";s:14:"res/apply.tmpl";s:4:"2711";s:26:"res/applyArrayFunction.php";s:4:"311c";s:15:"res/detail.tmpl";s:4:"edc1";s:26:"res/labelArrayFunction.php";s:4:"69f2";s:13:"res/list.tmpl";s:4:"c42b";s:27:"res/markerArrayFunction.php";s:4:"1606";s:12:"res/rss.tmpl";s:4:"9b41";s:15:"res/search.tmpl";s:4:"7397";s:27:"res/searchArrayFunction.php";s:4:"b96a";}',
	'suggests' => array(
	),
);

?>
